package.json

```diff
@@ -44,6 +44,7 @@
     "raw-loader": "^0.5.1",
     "sinon": "^7.1.1",
     "tape": "^4.9.1",
+    "tape-async": "^2.3.0",
```

Comment:
Is this needed?  We already have tests that return promises that work as expected.

Reply:
Yes, I had no success without this. We can't successfully `await` and return in the test. The current `await` that we do for the API in unit tests is not a valid test case since the promise resolves immediately.

Reply:
I don't think this is needed - I just tore it out as an experiment, and all of your tests (including new ones) pass.

I'm pretty sure this works because `tape` is async-by-default and tests don't end when the function returns - they only end when [`test.end()`](https://github.com/substack/tape#tenderr) is called or the [`t.plan()`](https://github.com/substack/tape#tplann)ned number of assertions occur.